internal/ansible/handler/logging_enqueue_annotation_test.go

```diff
@@ -0,0 +1,447 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/operator-lib/handler"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllertest"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+var _ = Describe("LoggingEnqueueRequestForAnnotation", func() {
+	var q workqueue.RateLimitingInterface
+	var instance LoggingEnqueueRequestForAnnotation
+	var pod *corev1.Pod
+	var podOwner *corev1.Pod
+
+	BeforeEach(func() {
+		q = controllertest.Queue{Interface: workqueue.New()}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "biz",
+				Name:      "biz",
+			},
+		}
+		podOwner = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "podOwnerNs",
+				Name:      "podOwnerName",
+			},
+		}
+
+		pod.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"})
+		podOwner.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"})
+
+		err := handler.SetOwnerAnnotations(podOwner, pod)
+		Expect(err).To(BeNil())
+		instance = LoggingEnqueueRequestForAnnotation{
+			handler.EnqueueRequestForAnnotation{
+				Type: schema.GroupKind{
+					Group: "",
+					Kind:  "Pod",
+				}}}
+	})
+
+	Describe("Create", func() {
+		It("should emit a log and enqueue a Request with the annotations of the object in case of CreateEvent", func() {
+			evt := event.CreateEvent{
+				Object: pod,
+			}
+
+			logBuffer.Reset()
```

Comment:
You can call `logBuffer.Reset()` in `BeforeEach()` (or `AfterEach()`) to avoid doing so in each `It()`.

Reply:
I had that initially but there were sometimes random log messages that would creep in from things that were running before the event handler method was called, and there are some tests where I check the log buffer multiple times, so just sticking the Reset before the actual call I want to check the log for seemed like the least finicky option